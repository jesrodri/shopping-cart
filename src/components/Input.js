import styled from "styled-components";

const borderColor = ({ theme, error }) => {
  if (error) {
    return theme.colors.input.border;
  } else {
    return theme.colors.input.error;
  }
}

const Input = styled.input`
  display: block;
  font: inherit;
  width: 100%;
  border-top-style: none;
  border-left-style: none;
  border-right-style: none;
  border-color: ${(props) => borderColor(props)};
  margin-block-start: ${({ theme }) => theme.space[3]};
  padding: 0;

  &:focus {
    outline: none;
  }

  option {
    color: ${({ theme }) => theme.colors.input.option};
  }
`

export default Input;
