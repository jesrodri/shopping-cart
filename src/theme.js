const colors = {
  black:       '#000000',
  white:       '#ffffff',
  lightGray:   '#b5b0a7',
  gray:        '#999999',
  darkGray:    '#37352f',
  darkRed:     '#cc0000',
  yellow:      '#e0ca00',
  lightYellow: '#f7e581',
  salmon:      '#fa8072',
  darkSalmon:  '#f16050',
  green:       '#66f57e',
  darkGreen:   '#49a659',
}

const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

const fonts = {
  body:
    "-apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica, 'Apple Color Emoji', Arial, sans-serif, 'Segoe UI Emoji', 'Segoe UI Symbol', 'Nunito', sans-serif;",
}

const fontSizes = [0, 5, 7, 9, 12, 16, 21, 28, 37, 50, 67, 89, 119, 159].map(
  (n) => `${n / 10}rem`
)

const fontWeights = [0, 300, 400, 700]

const radii = [0, 2]

const zIndices = []

const sizes = []

const theme = {
  colors,
  space,
  fonts,
  fontSizes,
  fontWeights,
  radii,
  zIndices,
  sizes,
}

export default theme
