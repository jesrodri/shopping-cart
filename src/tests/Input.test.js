import renderer from 'react-test-renderer';
import Input from '../components/Input';
import { ThemeProvider } from 'styled-components'
import theme from '../theme';

describe('<Input />', () => {
  it('renders Input correctly', () => {
    const input = renderer.create(
      <ThemeProvider theme={theme}>
        <Input />
      </ThemeProvider>
    ).toJSON();
    
    expect(input).toMatchSnapshot();
  });
});
