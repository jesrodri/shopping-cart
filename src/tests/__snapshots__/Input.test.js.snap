// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders Input correctly 1`] = `
<input
  className="Input-sc-1pydahm-0 iMhWWR"
/>
`;
